feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/SingleSelectDialog.kt

```diff
@@ -0,0 +1,91 @@
+package com.hedvig.android.odyssey.input.ui
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ListItem
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import coil.ImageLoader
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import org.koin.androidx.compose.get
+import com.hedvig.android.odyssey.R
+
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun <T> SingleSelectDialog(
+  title: String,
+  optionsList: List<T>,
+  onSelected: (T) -> Unit,
+  getDisplayText: (T) -> String,
+  getImageUrl: (T) -> String?,
+  onDismissRequest: () -> Unit,
+) {
+  val imageLoader: ImageLoader = get()
+
+  Dialog(onDismissRequest = { onDismissRequest.invoke() }) {
+    Surface(
+      modifier = Modifier.width(300.dp),
+      shape = RoundedCornerShape(10.dp),
+    ) {
+
+      Column(modifier = Modifier.padding(12.dp)) {
+
+        Text(text = title, Modifier.padding(12.dp))
+
+        Spacer(modifier = Modifier.height(12.dp))
+
+        LazyColumn {
+          items(
+            items = optionsList,
+            key = { option: T -> option.toString() },
```

Comment:
Are we 100% certain `toString()` on the option is going to be returning a unique string back no matter what? If not, it's better not to give a key and have it do extra recompositions rather than give the same key in two places accidentally and have it I think crash(?) due to that. Just gotta be a bit careful with these.

Reply:
Good catch. Ill ad a getId function as parameter